tests/unit-tests/internal/emails/test-class-email-generator.php

```diff
@@ -21,201 +21,68 @@ class Email_Generator_Test extends \WP_UnitTestCase {
-			]
-		);
-		$manage_url = esc_url(
-			add_query_arg(
-				array(
-					'page'      => 'sensei_learners',
-					'course_id' => $course->ID,
-					'view'      => 'learners',
-				),
-				admin_url( 'admin.php' )
-			)
-		);
-
-		( new Email_Generator() )->init();
-
-		$email_data = [
-			'name' => '',
-			'data' => null,
-		];
-
-		add_action(
-			'sensei_email_send',
-			function ( $email_name, $replacements ) use ( &$email_data ) {
-				$email_data['name'] = $email_name;
-				$email_data['data'] = $replacements;
-			},
-			10,
-			2
-		);
-
-		/* Act. */
-		do_action( 'sensei_user_course_start', $student_id, $course->ID );
+	protected $email_repository;
 
-		/* Assert. */
-		self::assertEquals( 'student_starts_course', $email_data['name'] );
-		self::assertArrayHasKey( 'test@a.com', $email_data['data'] );
-		self::assertEquals( 'Test Student', $email_data['data']['test@a.com']['student:displayname'] );
-		self::assertEquals( 'Test Course', $email_data['data']['test@a.com']['course:name'] );
-		self::assertEquals( $manage_url, $email_data['data']['test@a.com']['manage:students'] );
-	}
-
-	public function testGenerateEmail_WhenCalledByStudentCompletedCourseEvent_CallsEmailSendingActionWithRightData() {
-		/* Arrange. */
-		$student_id = $this->factory->user->create(
-			[
-				'display_name' => 'Test Student',
-			]
-		);
-		$teacher_id = $this->factory->user->create(
-			[
-				'user_email' => 'test@a.com',
-			]
-		);
-		$course     = $this->factory->course->create_and_get(
-			[
-				'post_title'  => 'Test Course',
-				'post_author' => $teacher_id,
-			]
-		);
-
-		$this->manuallyEnrolStudentInCourse( $student_id, $course->ID );
-
-		( new Email_Generator() )->init();
-
-		$email_data = [
-			'name' => '',
-			'data' => null,
-		];
-
-		add_action(
-			'sensei_email_send',
-			function ( $email_name, $replacements ) use ( &$email_data ) {
-				$email_data['name'] = $email_name;
-				$email_data['data'] = $replacements;
-			},
-			10,
-			2
-		);
-
-		/* Act. */
-		do_action( 'sensei_course_status_updated', 'complete', $student_id, $course->ID );
+	public function setUp(): void {
+		parent::setUp();
 
-		/* Assert. */
-		self::assertEquals( 'student_completes_course', $email_data['name'] );
-		self::assertArrayHasKey( 'test@a.com', $email_data['data'] );
-		self::assertEquals( 'Test Student', $email_data['data']['test@a.com']['student:displayname'] );
-		self::assertEquals( 'Test Course', $email_data['data']['test@a.com']['course:name'] );
-		self::assertArrayHasKey( 'manage:students', $email_data['data']['test@a.com'] );
-		self::assertNotEmpty( $email_data['data']['test@a.com']['manage:students'] );
+		$this->factory          = new Sensei_Factory();
+		$this->email_repository = new Email_Repository();
 	}
 
-	public function testGenerateEmail_WhenCalledByStudentUpdatedCourseEvent_DoesNotCallEmailIfCourseNotCompleted() {
+	public function testEmailGenerator_WhenInitIsCalled_AddsHooksForInitializingIndividualEmails() {
```

Comment:
```suggestion
	public function testInit_WhenCalled_AddsHooksForInitializingIndividualEmails() {
```
The whole test class tests the Email_Generator class, in this case there is no need to mention that in the test method name. (Same applies to other test methods.)